### ISSUE TITLE
Webp Tiles Appear with a blackboarder

### ISSUE THREAD

**Issue by paulsUsername** [mapbox/mapbox-maps-ios#2100]
Webp tiles pass through a tileset have a black boarder on them. They look fine in GL JS, but on the iOS SDK they have a large black boarder.

Any solution to this?

**Comment by OdNairy** [comment]
Hey @paulsUsername 
Can you provide a sample code to reproduce the issue and name the SDK version

**Comment by paulsUsername** [comment]
Hey!

Would you expect that the iOS SDK should support transparency from WebP tiles? 

The tile format we uploaded to Mapbox studio was WebP. Interestingly, if we upload in PNG format we don't have the issue.

So here is a screenshot from Mapbox studio of the map with transparency:

![Screenshot 2024-01-25 at 13 44 50](https://github.com/mapbox/mapbox-maps-ios/assets/32492218/780bab2e-4e5b-4afe-8f39-f22d5559079b)

Below is the style URL:

mapbox://styles/hiiker-mapbox/clrqlal5u00c501pl7wm0b02g (public)

Code to add to map (v11.1.0):

```
class ViewController: UIViewController {
    
    internal var mapView: MapView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        let centerCoordinate = CLLocationCoordinate2D(latitude: 53.5, longitude: -7.5)
        let camera = CameraOptions(
            center: centerCoordinate,
            zoom: 5
        )
        let options = MapInitOptions(
            cameraOptions: camera,
            styleURI: StyleURI(
                rawValue: "mapbox://styles/hiiker-mapbox/clrqlal5u00c501pl7wm0b02g"
            )
        )
        
        mapView = MapView(frame: view.bounds, mapInitOptions: options)
        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        
        view.addSubview(mapView)
    }
}
```


Screenshot results from the app:



![IMG_07EB06C910AC-1](https://github.com/mapbox/mapbox-maps-ios/assets/32492218/68d4e8c5-385a-425e-b0e3-1cbb86c485a3)

**Comment by paulsUsername** [comment]
@1ec5 One alternative is to use the static tiles API, but would I be correct in saying that you cannot download static tiles (due to lack of mapbox:// URL)?

**Comment by paulsUsername** [comment]
But also just for full clarity for anyone else, give the correct tileset (i.e. not a webp one) the iOS does support transparency:

![IMG_2025](https://github.com/mapbox/mapbox-maps-ios/assets/32492218/56831857-54e9-4a32-82de-66eaa7dbb289)

**Comment by OdNairy** [comment]
> WebP isn’t supported on iOS; you’re actually looking at a PNG the server returned instead.

iOS platform support WebP images since iOS 14. I was able to manually add WebP with an alpha channel with a transparent background. 
![IMG_0001](https://github.com/mapbox/mapbox-maps-ios/assets/735178/ae5103b0-041f-4c03-8795-f34ea78ba346)

Unfortunately, I've received 404 errors with @paulsUsername sample code.

**Comment by 1ec5** [comment]
Is that an image source connected to a server-side upload? Does the Raster Tiles API return WebP these days for iOS or just PNG? At least at one point, it used to return PNG because of missing WebP support on iOS.

**Comment by paulsUsername** [comment]
@1ec5 So if you are using the raster tiles API you can explicitly set the type (i.e. webp, png, png8) but when it is a mapbox scheme style, I don't know how to check (even the tile URLs are just mapbox://).

When using the style on the gl js framework, it does return webps as part of the style.

But the very strange thing is, If I upload my data to mapbox studio as PNGs it works perfectly with transparency. But if I upload a webp tileset (and add it to a style) transparency on the app does not work.

We have created a work around but uploading tilesets that are a mix of png and jpeg (which seems to cause no issues). Where there is transparency we leave the tile as a PNG, where there is no transparency we convert to JPEG.

So you get the benefit of small uploadable tilesets and pretty much all of the benefit of WebP (reduced file size).

**Comment by OdNairy** [comment]
@paulsUsername yep, I've added a local WebP to check if renderer supports WebP format in general. 
However, @ 1ec5 is right, and server converts WebP into PNG (JPEG?). 
I recommend you stick to the workaround you've found already while we investigate the actual WebP support.

**Comment by 1ec5** [comment]
Oh yeah, it’s probably JPEG, now that you mention it. That sounds like a neat workaround by the way.
